File: CS50x/sentimental-readability/readability.py
from cs50 import get_string

text = get_string('Text: ')

word = 1
letter = 0
sentence = 0
for i in text:
    letter = letter + 1
    if i == '.' or i == '!' or i == '?':
        letter = letter - 1
        sentence = sentence + 1
    if i == ' ':
        letter = letter - 1
        word = word + 1
    if i == ',' or i == ':' or i == ';' or i == "'":
        letter = letter - 1

L = (letter / word) * 100
S = (sentence / word) * 100


index = round(0.0588 * L - 0.296 * S - 15.8)


if index < 1:
    index = 'Before Grade 1'
elif index >= 16:
    index = 'Grade 16+'
else:
    index = f'Grade {index}'


print(index)

File: CS50x/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    float w = 1;
    float l = 0;
    float s = 0;
    float letter_average = 0;
    float sentence_average = 0;
    float index = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            w++;
        }

        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            s++;
        }

        else if (tolower(text[i]) >= 'a' && tolower(text[i]) <= 'z')
        {
            l++;
        }
    }

    sentence_average = ((s / w) * 100);
    letter_average = (((l) / w) * 100);

    index = 0.0588 * letter_average - 0.296 * sentence_average - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}
